src/pages/StocktakesPage.js

```diff
@@ -8,44 +8,32 @@
 import React, { useCallback } from 'react';
 import PropTypes from 'prop-types';
 import { View } from 'react-native';
-import { SearchBar } from 'react-native-ui-components';
 
-import { MODAL_KEYS, getAllPrograms } from '../utilities';
-import { buttonStrings, modalStrings } from '../localization';
 import { UIDatabase } from '../database';
 import Settings from '../settings/MobileAppSettings';
+
+import { MODAL_KEYS, getAllPrograms } from '../utilities';
+import { usePageReducer, useSyncListener, useNavigationFocus } from '../hooks';
+import { getItemLayout, recordKeyExtractor } from './dataTableUtilities';
+
+import { PageButton, DataTablePageView, SearchBar } from '../widgets';
 import { BottomConfirmModal, DataTablePageModal } from '../widgets/modals';
-import { PageButton } from '../widgets';
 import { DataTable, DataTableHeaderRow, DataTableRow } from '../widgets/DataTable';
-import {
-  sortData,
-  filterData,
-  selectRow,
-  deselectRow,
-  deselectAll,
-  closeBasicModal,
-  deleteRequisitions,
-  openBasicModal,
-} from './dataTableUtilities/actions';
-import { getItemLayout, recordKeyExtractor } from './dataTableUtilities/utilities';
-import globalStyles, { SUSSOL_ORANGE, newDataTableStyles, newPageStyles } from '../globalStyles';
-import usePageReducer from '../hooks/usePageReducer';
-import DataTablePageView from './containers/DataTablePageView';
+
+import { buttonStrings, modalStrings } from '../localization';
+import globalStyles, { SUSSOL_ORANGE, newPageStyles } from '../globalStyles';
 
 import {
   gotoStocktakeManagePage,
   createStocktake,
   gotoStocktakeEditPage,
 } from '../navigation/actions';
 
-const keyExtractor = item => item.id;
-
-export const StocktakesPage = ({ routeName, currentUser, dispatch: reduxDispatch }) => {
-  const [state, dispatch, instantDebouncedDispatch, debouncedDispatch] = usePageReducer(routeName, {
-    backingData: UIDatabase.objects('Stocktake'),
-    data: UIDatabase.objects('Stocktake')
-      .sorted('createdDate', false)
-      .slice(),
+const stateInitialiser = () => {
```

Comment:
Potentially put state initialisers for all pages in a file which can be used by each pages reducer for an easy way to 'reset' state

Reply:
that's an interesting idea. it would make components smaller and easier to read which I like; just whether it means when changing a component you have to refer to multiple files and whether that is a bother. I think overall, good idea

Reply:
You're right this is the way I was thinking also - The problem being at this point is that the `initialState` is sometimes derived from the props passed to the page, so there needs to be some parameter passing here. 

The impurity is the main reason I haven't done it. Yet to think of a good solution which enables the use of the initialiser function *as well* as having it usable within the reducer